zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -1074,11 +1125,53 @@ where
                 }
             }
             PoolType::Transparent => {
+                #[cfg(not(feature = "transparent-inputs"))]
                 return Err(Error::UnsupportedChangeType(output_pool));
             }
         }
     }
 
+    // This reserves the ephemeral addresses even if transaction construction fails.
+    // It is not worth the complexity of being able to unreserve them, because there
+    // are few failure modes after this point that would allow us to do so.
+    #[cfg(feature = "transparent-inputs")]
+    {
+        let ephemeral_outputs: Vec<(usize, &ChangeValue)> = proposal_step
+            .balance()
+            .proposed_change()
+            .iter()
+            .enumerate()
+            .filter(|(_, change_value)| matches!(change_value.output_pool(), PoolType::Transparent))
```

Comment:
Is the pool type all that we have to go on to determine that this is indeed supposed to be an interstitial output? While `SingleChangeOutputStrategy` behaves correctly here, I'm not sure we can be confident that this is the case for arbitrary transparent change outputs.

Reply:
Now that `ChangeValue` has an `is_ephemeral` flag, we should be checking that here.